solutions/silver/cf-1744F.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: cf-1744F
+source: CF
+title: MEX vs MED
+author: Alex Du
+---
+
+<Spoiler title="Hint 1">
+
+To find subsegments with a given value of $\texttt{mex}$, we must find subsegments which contain all
+integers $0, 1, \ldots, \texttt{mex}-1$, while also not containing $\texttt{mex}$.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+For a subsegment of length $\leq 2 \cdot \texttt{mex}$, the condition $\texttt{mex} > \texttt{med}$ will be satisfied, since the median can be at most $\texttt{mex}-1$.
+
+</Spoiler>
+
+<Spoiler title="Solution">
+
+## Explanation
+
+[Official Analysis](https://codeforces.com/blog/entry/108101)
+
+We will iterate $\texttt{mex}$ from $1$ to $n-1$, finding all valid subsegments of that $\texttt{mex}$ at a time.
+
+Let $l$ and $r$ represent the highest left bound and the lowest right bound respectively of a subsegment that satisfies a given $\texttt{mex}$, and $\texttt{pos}$ represent  the index of $\texttt{mex}$. If we know $l$, $r$, and $\texttt{pos}$, then we can quickly count the subsegments for a given $\texttt{mex}$.
+
+First, if $\texttt{pos}$ is between $l$ and $r$, the number of valid subsegments with this $\texttt{mex}$ is zero since the $\texttt{mex}$ of this subsegment is greater than the $\texttt{mex}$ we are looking for.
+
+We need to find all subsegments of length $\leq 2 \cdot \texttt{mex}$ that contain $l \ldots r$ and don't include $\texttt{pos}$. Say $\texttt{pos}>r$, since we only need to change our approach slightly if $\texttt{pos} < l$. We can increment $r$ from $r$ to $\texttt{pos}-1$ fast enough, since each time we increment we increase the size of our subsegment by $1$, for a maximum of $N$ times.
+
+Considering this new bound of $r$, we can find the number of possible values of $l$ that form a valid subsegment. We know that the $\texttt{mex}$ is correct for this subsegment, so we just need to find the number of positions $\leq l$ such that $r-l+1 \leq \texttt{mex} \cdot 2$. The lower
+bound of our smallest value of $l$ is $r-\texttt{mex} \cdot 2 + 1$, which we can find by rearranging the previous inequality.
+
+Note that if this value is negative, we should set it to $0$. After this, we know that all values between $l$ and this smallest $l$ are valid left bounds for our given $r$ so that we can add $l-\texttt{smallest\_l} + 1$. Again, note that if $l < \texttt{smallest\_l}$, which will happen in the case that our subsegment is too big for our current $\texttt{mex}$, we should instead add $0$.
```

Comment:
```suggestion
After this, we know that all values between $l$ and the lower bound of $l$ are valid left bounds for our given $r$, so we can add $l-\texttt{smallest\_l} + 1$ to the answer.
```

Reply:
Add the original third sentence (again, note that...we should instead add $0$) to the second sentence but make it concise and not its own sentence